File: articles/feedback-loops-delays-and-policy-resistance/julia/policy_resistance_model.jl
# policy_resistance_model.jl

function net_policy_effect(policy_delta, intended_strength, resistance_strength, resistance_response)
    return intended_strength * policy_delta - resistance_strength * resistance_response
end

println("Net policy effect = ", round(net_policy_effect(10.0, 0.8, 0.4, 6.0), digits=6))

File: articles/feedback-loops-delays-and-policy-resistance/python/policy_resistance_model.py
#!/usr/bin/env python3
"""Policy resistance helper."""

from __future__ import annotations


def net_policy_effect(policy_delta: float, intended_strength: float, resistance_strength: float, resistance_response: float) -> float:
    return intended_strength * policy_delta - resistance_strength * resistance_response


if __name__ == "__main__":
    print(round(net_policy_effect(10.0, 0.8, 0.4, 6.0), 6))
